.github/workflows/run-test-cases.yml

```diff
@@ -127,172 +137,182 @@ jobs:
             kube-version: 1.20.1-00
           - test-case: end-to-end
             test-file: test/run-end-to-end.py
+          - test-case: webhook
```

Comment:
webhook may need to appear in the matrix.test-case section as well, i'm not sure

Reply:
i created a test repo (https://github.com/bfjelds/job-strategy-test) and it looks like you don't get a full everything-by-everything matrix if there is an item of type foo in the `matrix.include` section that isn't in the `matrix.foo` section ... i think we need webhook to appear here:

```yaml
    strategy:
      matrix:
        test-case:
          - webhook
```

Reply:
Awesome.

Thank you very much.

Headed home now but to cook.

Will fix tomorrow unless I've broken you too?

No snow here yet!

Reply:
Nothing is broken, enjoy dinner!

Reply:
Continued to fail so added this too: https://github.com/deislabs/akri/pull/206/commits/bf8e62cdff3d48c7ed3d97962ffe010f29182c71

Reply:
Could this `matrix` not simplified?

```YAML
matrix:
  kube:
  - runtime: K3s-1.18
    version: 1.18/stable
  - runtime: K3s-1.19
    version:
   - ...
  test-case:
  - case: end-to-end
    file: test/run-end-to-end.py
  - case: webhook
    file: test/run-webhook.py
```

IIUC `include` provides a mechanism to add exceptional jobs; that aren't for-all (better explanation escapes me):

i.e. if we wanted (for whatever reason to not run `end-to-end` but to run `webhook` on runtime K3s-1.20, we could:

```YAML
matrix:
  kube:
    ...
  test-case:
    ...
  include:
  - runtime: K3s-1.20
    version: 1.20/stable
    case: webhook
    file: test/run-webhook.py
```

Reply:
brilliant.  i love this.

just validated it ... this works perfectly:

```yaml
  strategy-test:
    strategy:
      matrix:
        kube:
          - runtime: K3s-1.18
            version: v1.18.9+k3s1
          - runtime: K3s-1.19
            version: v1.19.4+k3s1
          - runtime: K3s-1.20
            version: 1.20/stable
        test:
          - case: end-to-end
            file: test/run-end-to-end.py
          - case: webhook
            file: test/run-webhook.py

    steps:
      - name: echo strategy
        run: |
          echo "kube-runtime: " ${{ matrix.kube.runtime }}
          echo "kube-version: " ${{ matrix.kube.version }}
          echo "test-case:    " ${{ matrix.test.case }}
          echo "test-file:    " ${{ matrix.test.file }}
```